src/test/java/com/google/growpod/servlets/GardenServletTest.java

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import com.google.growpod.controllers.GardenController;
+import com.google.growpod.data.Garden;
+import com.google.growpod.servlets.GardenServlet;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+/**
+ * Tests Garden servlet response behavior, based on different request URL and controller responses.
+ */
+@ExtendWith(MockitoExtension.class)
+public final class GardenServletTest {
+
+  private GardenServlet servlet; // Class to test
+
+  /** Mock services. */
+  @Mock private GardenController controller;
+
+  /** Mock values. */
+  private final Garden TEST_GARDEN = new Garden("0", "x", 0.0, 0.0, "0");
+  /** Separate lists in case I change the type each query returns */
+  private final List<String> TEST_USER_LIST = Arrays.asList("0");
+
+  private final List<String> TEST_PLANT_LIST = Arrays.asList("0");
+
+  /** Initializes servlet object and mock controller. */
+  @BeforeEach
+  public void initTest() {
+    servlet = new GardenServlet();
+    servlet.setController(controller);
+  }
+
+  /** Tests successful query for GET: /garden/{id} method. */
+  @Test
+  public void testGetGarden() throws IOException {
```

Comment:
it is pretty silly to name all tests `testXYZ`. We know it is a test. The file says test all over it as well as the test is decorated with it :).

In general in Java, the best way to name tests are as follows:
`functionUnderTest_precondition_postcondition`.

In this case, that would mean: `doGet_validGardenId_successfulResult`.
Another example from further down would be: `doGet_missingGardenId_returns404`.

Reply:
Will change.